lib/matplotlib/backend_bases.py

```diff
@@ -1595,6 +1595,43 @@ def __init__(self, figure):
         self.toolbar = None  # NavigationToolbar2 will set me
         self._is_idle_drawing = False
 
+    def _repr_html_(self):
+        from base64 import b64encode
+
+        fmt = self.get_default_filetype()
+        dpi = self.figure.dpi
+        if fmt == 'retina':
+            dpi = dpi * 2
+            fmt = 'png'
+
+        kw = {
+            "format":fmt,
+            "facecolor":self.figure.get_facecolor(),
+            "edgecolor":self.figure.get_edgecolor(),
+            "dpi":dpi,
+            "bbox_inches":self.figure.bbox_inches,
+        }
+
+        bytes_io = io.BytesIO()
+        self.print_figure(bytes_io, **kw)
+        raw_bytes = bytes_io.getvalue()
+
+        mimetype_dict = {'png': 'image/png', 'svg': 'image/svg+xml', 
+                         'jpg': 'image/jpeg', 'pdf': 'application/pdf'}
+        if fmt not in mimetype_dict:
```

Comment:
falling back to png *here* seems wrong/too late (it could well not be a png file, indeed).

Reply:
As of now, running %matplotlib inline uses default figure_format as png. So regardless of the backend, a png will be produced.